### PR TITLE
Added document to use TLS Secured YugabyteDB Cluster by Helm Charts

### PR DESCRIPTION
**Pull request by baba230896**
Added document to use TLS Secured YugabyteDB Cluster by Helm Charts

**Added two approaches to connect the TLS Secured YugabyteDB cluster -** 

1. Within Kubernetes Cluster.
2. Remotely.

Fixes: https://github.com/yugabyte/yugabyte-db/issues/4726

### CODE REVIEW

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 1]
```diff
@@ -109,3 +109,137 @@ Status:
         router: no errors
         loadbalancer emulator: no errors
 ```
+
+## Connecting TLS Secured YugabyteDB cluster deployed by Helm Charts
+
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:114]: We should link to the helm deployment doc from here and say `To bring up a yugabyte server with encryption in transit (TLS) enabled, set the flag tls.enabled = true in the helm overrides`

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 2]
```diff
@@ -109,3 +109,137 @@ Status:
...
+```
+
+(Optional) Use the below-mentioned command to remove the `yb-client` pod.
+
+```sh
+$ kubectl delete po yb-client -n yb-demo
+pod "yb-client" deleted
+```
+
+### Connecting Remote Cluster
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:189]: `Connecting externally` might be a better name for this section.

File: docs/content/latest/deploy/kubernetes/clients.md (hunk 1 above)
Comment (baba230896) [reply docs/content/latest/deploy/kubernetes/clients.md:114]: Okay, I'll update the suggested changes.

File: docs/content/latest/deploy/kubernetes/clients.md (hunk 2 above)
Comment (baba230896) [reply docs/content/latest/deploy/kubernetes/clients.md:189]: Yeah, this one is comparatively better. I'll update it.

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 3]
```diff
@@ -109,3 +109,139 @@ Status:
         router: no errors
         loadbalancer emulator: no errors
 ```
+
+## Connecting TLS Secured YugabyteDB cluster deployed by Helm Charts
+
+Follow the following [documentation link](https://docs.yugabyte.com/latest/deploy/kubernetes/single-zone/gke/helm-chart/) for the deployment of YugabyteDB cluster using Helm Chart and to bring up the cluster with encryption in transit (TLS) enabled, set the flag `tls.enabled=true` in the helm command-line.
+
+### Within Kubernetes cluster
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:117]:
```suggestion
### Connecting from within the Kubernetes cluster
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 4]
```diff
@@ -109,3 +109,139 @@ Status:
         router: no errors
         loadbalancer emulator: no errors
 ```
+
+## Connecting TLS Secured YugabyteDB cluster deployed by Helm Charts
+
+Follow the following [documentation link](https://docs.yugabyte.com/latest/deploy/kubernetes/single-zone/gke/helm-chart/) for the deployment of YugabyteDB cluster using Helm Chart and to bring up the cluster with encryption in transit (TLS) enabled, set the flag `tls.enabled=true` in the helm command-line.
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:115]:
```suggestion
To bring up a YugabyteDB cluster with encryption in transit (TLS) enabled, follow the [instructions here](https://docs.yugabyte.com/latest/deploy/kubernetes/single-zone/gke/helm-chart/) and set the flag `tls.enabled=true` in the helm command-line.

For example, `helm install yugabyte --namespace yb-demo --name yb-demo --set=tls.enabled=true`.
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 5]
```diff
@@ -109,3 +109,137 @@ Status:
...
+    secret:
+      secretName: yugabyte-tls-client-cert
+      defaultMode: 256
+```
+
+Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect.
+
+Use the following command to verify the connection.
+
+```sh
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:148]: We should combine these two command examples into a single command example which connects via ysqlsh and shows the result of running `\conninfo` within that shell.

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 6]
```diff
@@ -109,3 +109,137 @@ Status:
...
+
+yugabyte=#
+```
+
+Here is an example of a client that uses the `YCQL shell` ([`ycqlsh`](../../../admin/cqlsh)) to connect.
+
+Use the following command to verify the connection.
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ycqlsh yb-tservers.yb-demo.svc.cluster.local 9042 --ssl --execute 'SHOW HOST'
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:170]: Same here, let's combine these into a single command example.

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 7]
```diff
@@ -109,3 +109,137 @@ Status:
...
+Connected to local cluster at yb-tservers.yb-demo.svc.cluster.local:9042.
+[cqlsh 5.0.1 | Cassandra 3.9-SNAPSHOT | CQL spec 3.4.2 | Native protocol v4]
+Use HELP for help.
+cqlsh>
+```
+
+(Optional) Use the below-mentioned command to remove the `yb-client` pod.
+
+```sh
+$ kubectl delete po yb-client -n yb-demo
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:187]:
```suggestion
$ kubectl delete pod yb-client -n yb-demo
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 8]
```diff
@@ -109,3 +109,137 @@ Status:
...
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ycqlsh yb-tservers.yb-demo.svc.cluster.local 9042 --ssl
+Connected to local cluster at yb-tservers.yb-demo.svc.cluster.local:9042.
+[cqlsh 5.0.1 | Cassandra 3.9-SNAPSHOT | CQL spec 3.4.2 | Native protocol v4]
+Use HELP for help.
+cqlsh>
+```
+
+(Optional) Use the below-mentioned command to remove the `yb-client` pod.
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:184]:
```suggestion
(Optional) After the operations are complete, remove the client pod. 
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 9]
```diff
@@ -109,3 +109,137 @@ Status:
...
+```sh
+$ mkdir $(pwd)/certs
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.root\.crt}' | base64 --decode > $(pwd)/certs/root.crt
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.yugabytedb\.crt}' | base64 --decode > $(pwd)/certs/yugabytedb.crt
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.yugabytedb\.key}' | base64 --decode > $(pwd)/certs/yugabytedb.key
+```
+
+Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect.
+
+Use the following command to verify the connection.
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:208]: Same as above: let's collapse these two examples into one which connects and shows the output of \conninfo

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 10]
```diff
@@ -109,3 +109,137 @@ Status:
...
+(Optional) Use the below-mentioned command to remove the `yb-client` pod.
+
+```sh
+$ kubectl delete po yb-client -n yb-demo
+pod "yb-client" deleted
+```
+
+### Connecting Remote Cluster
+
+Prerequisites:
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:193]: Remove the prerequisites step.

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 11]
```diff
@@ -109,3 +109,137 @@ Status:
...
+1. Client Certificates
+2. External Cluster IP
+
+Use the following commands to get the client certificates from Kubernetes cluster's Secrets.
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:197]:
```suggestion
To connect externally to a TLS-enabled YugabyteDB helm cluster, first download the client certificates locally from the Kubernetes cluster's secrets.
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 12]
```diff
@@ -109,3 +109,137 @@ Status:
...
+
+```sh
+$ mkdir $(pwd)/certs
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.root\.crt}' | base64 --decode > $(pwd)/certs/root.crt
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.yugabytedb\.crt}' | base64 --decode > $(pwd)/certs/yugabytedb.crt
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.yugabytedb\.key}' | base64 --decode > $(pwd)/certs/yugabytedb.key
+```
+
+Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect.
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:206]:
```suggestion
Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect. The command specifies the external LoadBalancer IP of the `yb-tserver-service` as shown in [the docs here](../single-zone/oss/helm-chart/#connect-using-external-clients). 
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 13]
```diff
@@ -109,3 +109,137 @@ Status:
...
+ysqlsh (11.2-YB-2.1.5.0-b0)
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+Type "help" for help.
+
+yugabyte=#
+```
+
+Here is an example of a client that uses the `YCQL shell` ([`ycqlsh`](../../../admin/cqlsh)) to connect.
+
+Use the following docker command to verify the connection.
```
Comment (iSignal) [inline docs/content/latest/deploy/kubernetes/clients.md:229]: Same as above, let's collapse into a single example.

File: docs/content/latest/deploy/kubernetes/clients.md (hunk 5 above)
Comment (baba230896) [reply docs/content/latest/deploy/kubernetes/clients.md:150]: Sure, Will do

File: docs/content/latest/deploy/kubernetes/clients.md (hunk 10 above)
Comment (baba230896) [reply docs/content/latest/deploy/kubernetes/clients.md:195]: okay.

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 14]
```diff
@@ -80,7 +80,7 @@ ycqlsh> use demo;
 ycqlsh:demo> CREATE TABLE t_demo(id INT PRIMARY KEY);
 ```
 
-### Master UI dashboard
+## Master UI dashboard
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:83]:
```suggestion
## YB-Master Admin UI
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 15]
```diff
@@ -80,7 +80,7 @@ ycqlsh> use demo;
...
 
 The master UI dashboard is available at the external IP address exposed by the `yb-master-ui` LoadBalancer service - in this case at `http://98.138.219.231:7000/`.
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:85]:
```suggestion
The YB-Master Admin UI is available at the IP address exposed by the `yb-master-ui` LoadBalancer service – at `https://98.138.219.231:7000/`.
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 16]
```diff
@@ -109,3 +109,112 @@ Status:
         router: no errors
         loadbalancer emulator: no errors
 ```
+
+## Connecting TLS Secured YugabyteDB cluster deployed by Helm Charts
+
+To bring up a YugabyteDB cluster with encryption in transit (TLS) enabled, follow the [instructions here](https://docs.yugabyte.com/latest/deploy/kubernetes/single-zone/gke/helm-chart/) and set the flag `tls.enabled=true` in the helm command-line.
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:115]:
```suggestion
To start a YugabyteDB cluster with encryption in transit (TLS) enabled, follow the steps at [Google Kubernetes Service - Helm Chart](https://docs.yugabyte.com/latest/deploy/kubernetes/single-zone/gke/helm-chart/) and set the flag `tls.enabled=true` in the helm command-line.
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 17]
```diff
@@ -109,3 +109,112 @@ Status:
...
+
+For example, `helm install yugabyte --namespace yb-demo --name yb-demo --set=tls.enabled=true`.
+
+### Connecting from within the Kubernetes cluster
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:119]:
```suggestion
### Connect from within the Kubernetes cluster
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 18]
```diff
@@ -109,3 +109,112 @@ Status:
...
+
+Copy the following `yb-client.yaml` and use this `kubectl create -f yb-client.yaml` command to create a pod with auto-mounted client certificates.
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:121]:
```suggestion
Copy the following `yb-client.yaml` and use the `kubectl create -f yb-client.yaml` command to create a pod with auto-mounted client certificates.
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 19]
```diff
@@ -109,3 +109,112 @@ Status:
...
+Use the following command to verify the connection.
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ysqlsh -h yb-tservers.yb-demo.svc.cluster.local "sslmode=require"
+ysqlsh (11.2-YB-2.1.5.0-b0)
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+Type "help" for help.
+
+yugabyte=# \conninfo
+You are connected to database "yugabyte" as user "yugabyte" on host "yb-tservers.yb-demo.svc.cluster.local" at port "5433".
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:157]:
```suggestion
You are connected to the `yugabyte` database as user `yugabyte` on host `yb-tservers.yb-demo.svc.cluster.local` at port `5433`.
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 20]
```diff
@@ -109,3 +109,112 @@ Status:
...
+```
+
+(Optional) After the operations are complete, remove the client pod. 
+
+```sh
+$ kubectl delete pod yb-client -n yb-demo
+pod "yb-client" deleted
+```
+
+### Connecting externally
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:181]:
```suggestion
### Connect externally
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 21]
```diff
@@ -109,3 +109,112 @@ Status:
...
+To connect externally to a TLS-enabled YugabyteDB helm cluster, first download the client certificates locally from the Kubernetes cluster's secrets.
+
+```sh
+$ mkdir $(pwd)/certs
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.root\.crt}' | base64 --decode > $(pwd)/certs/root.crt
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.yugabytedb\.crt}' | base64 --decode > $(pwd)/certs/yugabytedb.crt
+$ kubectl get secret yugabyte-tls-client-cert  -n yb-demo -o jsonpath='{.data.yugabytedb\.key}' | base64 --decode > $(pwd)/certs/yugabytedb.key
+```
+
+Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect. The command specifies the external LoadBalancer IP of the `yb-tserver-service` as shown in [the docs here](../single-zone/oss/helm-chart/#connect-using-external-clients).
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:192]:
```suggestion
Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect. The command specifies the external LoadBalancer IP of the `yb-tserver-service` as described in [Connect using external clients](../single-zone/oss/helm-chart/#connect-using-external-clients). 
```

File: docs/content/latest/deploy/kubernetes/clients.md [hunk 22]
```diff
@@ -109,3 +109,112 @@ Status:
...
+Type "help" for help.
+
+yugabyte=# \conninfo
+You are connected to database "yugabyte" as user "yugabyte" on host "35.200.205.208" at port "5433".
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+```
+
+Here is an example of a client that uses the `YCQL shell` ([`ycqlsh`](../../../admin/cqlsh)) to connect.
+
+Use the following docker command to verify the connection.
```
Comment (stevebang) [inline docs/content/latest/deploy/kubernetes/clients.md:209]:
```suggestion
To verify the connection, use the following `docker run` command.
```

File: docs/content/latest/deploy/kubernetes/clients.md (hunk 19 above)
Comment (baba230896) [reply docs/content/latest/deploy/kubernetes/clients.md:157]: This can't be edited as this is the output of the `\conninfo` command.

Comment (stevebang) [reply docs/content/latest/deploy/kubernetes/clients.md:157]: I missed that -- OK as-is then.

**Pull request merged**